drivers/ethernet/Kconfig.stm32_hal

```diff
@@ -56,6 +56,12 @@ config ETH_STM32_HAL_RX_THREAD_PRIO
 	  Preemptive scheduling can lead to more responsive handling of network traffic,
 	  especially under high load.
 
+config ETH_STM32_HAL_ETH_POST_INIT_PRIO
+	int "Second part of the ethernet initialization"
+	default 89
```

Comment:
I'm a little bit worried that this doesn't take `ETH_INIT_PRIORITY` into account and likely anyone moving `ETH_INIT_PRIORITY` will miss to update `ETH_STM32_HAL_ETH_POST_INIT_PRIO`.
At the very least it should mention the dependency.

Also, it doesn't seem to be compatible  with `ETH_STM32_HAL_PTP_CLOCK_INIT_PRIO` defined at 85 by default and which should be higher than this one.

Reply:
`89` should still be fixed as it can't be higher than `ETH_STM32_HAL_PTP_CLOCK_INIT_PRIO` (`85` today).

Also, question on MDIO priority which is `CONFIG_ETH_INIT_PRIORITY` today. How do we ensure is is done after ethernet early initialization phase ?